File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        if (k == len(nums)):
            return nums

        counterMap = {}
        resMap = defaultdict(list)
        for num in nums:
            counterMap[num] = 1 + counterMap.get(num, 0)

        for key, value in counterMap.items():
            resMap[value].append(key)

        freqList = list(resMap.keys())
        freqList.sort(reverse=True)
        res = []
        counter = 0
        for key in freqList:
            for num in resMap[key]:
                if (len(res) < k):
                    res.append(num)
                    counter += 1
        return res

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-3.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        vector<int> counts(2001, 0);

        for (int& n: nums) { counts[n+1000]++; }

        // for (int i = 0; i < 2001;i++) {cout << counts[i] << " ";}

        int r = 2000; int count = 1;

        while (r >= 0) {
            while (counts[r] > 0) {
                if (count == k) { return r -1000; }
                count++;
                counts[r]--;
            }
            r -= 1;
        }

        return 0;
    }
};
